docs/references/nextjs/auth-middleware.mdx

```diff
@@ -36,20 +73,57 @@ import { authMiddleware, redirectToSignIn } from '@clerk/nextjs';
 
 export default authMiddleware({
   afterAuth(auth, req, evt) {
-    // handle users who aren't authenticated
+    // Handle users who aren't authenticated
     if (!auth.userId && !auth.isPublicRoute) {
       return redirectToSignIn({ returnBackUrl: req.url });
     }
-    // redirect them to organization selection page
+    // Redirect logged in users to organization selection page if they are not active in an organization
     if(auth.userId && !auth.orgId && req.nextUrl.pathname !== "/org-selection"){
       const orgSelection = new URL('/org-selection', req.url)
       return NextResponse.redirect(orgSelection)
     }
+    // If the user is logged in and trying to access a protected route, allow them to access route
+    if (auth.userId && !auth.isPublicRoute) {
+      return NextResponse.next()
+    }
+    // Allow users visiting public routes to access them
+    return NextResponse.next();
   }
 });
 ```
 
-### Using `beforeAuth()` to execute middleware before authentication
+#### Understanding `afterAuth` configuration
+
+When using `afterAuth`, the checks must handle all cases — a custom `afterAuth` completely overrides the default behavior covered earlier, aside from adding sign-in and sign-up to `publicRoutes`. `afterAuth` does not allow you to add just one check and rely on the default behavior otherwise. 
+
+Looking at the example immediately above, the following is happening:
+
+1. Redirect all users who are not logged in but who are visiting a protected page to the sign-in route.
+```ts 
+if (!auth.userId && !auth.isPublicRoute) {
+      return redirectToSignIn({ returnBackUrl: req.url });
+    }
+```
+2. If the user is signed in but is not currently active in an organization and is not visiting the route for selecting an org, then redirect them to the organization selector. This will force users to select and be active in an organization to visit other parts of the site.
```

Comment:
```suggestion
2. If the user is signed in but is not currently active in an organization and is not visiting the route for selecting an organization, then redirect them to the organization selector. This will force users to select and be active in an organization to visit other parts of the site.
```